core/src/test/java/org/infinispan/statetransfer/StateConsumerTest.java

```diff
@@ -196,95 +190,243 @@ public void test1() throws Exception {
       // number of in-flight requests will decrease accordingly. During real usage, the state chunk collections can
       // have more than a single segment.
       long inflightCounter = requestedSegments.values().stream().mapToLong(Collection::size).sum();
       assertEquals(stateConsumer.inflightRequestCount(), inflightCounter);
       for (Map.Entry<Address, Set<Integer>> entry : requestedSegments.entrySet()) {
          for (Integer segment : entry.getValue()) {
-            Collection<StateChunk> chunks = Collections.singletonList(new StateChunk(segment, Collections.emptyList(), true));
-            stateConsumer.applyState(entry.getKey(), 4, chunks)
+            Collection<StateChunk> chunks = Collections.singletonList(
+                  new StateChunk(segment, Collections.emptyList(), true));
+            stateConsumer.applyState(entry.getKey(), topologyId, chunks)
                   .toCompletableFuture()
                   .get(10, TimeUnit.SECONDS);
 
             inflightCounter -= 1;
             assertEquals(stateConsumer.inflightRequestCount(), inflightCounter);
          }
       }
+      assertFalse(stateConsumer.hasActiveTransfers());
+      assertEquals(stateConsumer.inflightRequestCount(), 0);
+   }
+
+   private static void applyState(StateConsumer stateConsumer, Map<Address, Set<Integer>> requestedSegments, Collection<InternalCacheEntry<?, ?>> cacheEntries) {
+      Map.Entry<Address, Set<Integer>> entry = requestedSegments.entrySet().iterator().next();
+      Collection<StateChunk> chunks = Collections.singletonList(
+            new StateChunk(entry.getValue().iterator().next(), cacheEntries, true));
+      stateConsumer.applyState(entry.getKey(), 22, chunks);
+   }
+
+   private void injectComponents(StateConsumer stateConsumer, AsyncInterceptorChain interceptorChain, RpcManager rpcManager) {
+      TestingUtil.inject(stateConsumer,
+            mockCache(),
+            TestingUtil.named(NON_BLOCKING_EXECUTOR, pooledExecutorService),
+            mock(LocalTopologyManager.class),
+            interceptorChain,
+            mockInvocationContextFactory(),
+            createConfiguration(),
+            rpcManager,
+            mockCommandsFactory(),
+            mockPersistenceManager(),
+            mock(InternalDataContainer.class),
+            mockTransactionTable(),
+            mock(StateTransferLock.class),
+            mock(CacheNotifier.class),
+            new CommitManager(),
+            new CommandAckCollector(),
+            new TriangleOrderManager(0),
+            new HashFunctionPartitioner(),
+            mock(InternalConflictManager.class),
+            mock(DistributionManager.class),
+            mock(LocalPublisherManager.class),
+            mock(PerCacheInboundInvocationHandler.class),
+            mockXSiteStateTransferManager());
+   }
+
+   public void test1() throws Exception {
```

Comment:
It would be good to give both tests a more meaningful name if possible, as well as providing a high-level description of the tests purpose as a comment on the test method.

Reply:
That's much better, thanks :+1: